src/header.rs

```diff
@@ -592,10 +721,14 @@ impl fmt::Display for TbfHeader {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         write!(f, "TBF Header:")?;
         write!(f, "{}", self.hdr_base)?;
-        write!(f, "{}", self.hdr_main)?;
+        write!(f, "{:?}", self.hdr_main)?;
```

Comment:
```suggestion
                self.hdr_main
            .map_or(Ok(()), |hdr| write!(f, "{}", hdr))?;
```

Otherwise it prints it like:

```
Some(TbfHeaderMain { base: TbfHeaderTlv { tipe: Main, length: 12 }, init_fn_offset: 41, protected_size: 0, minimum_ram_size: 4604 })
```

Reply:
Done.